src/plugin.js

```diff
@@ -157,4 +159,40 @@
     isArray: Array.isArray
   };
 
+  // Polyfill the tech function with properties for the current tech. This is
+  // a mostly-complete implementation of the old tech behavior.
+  Player.prototype.loadTech_ = function() {
+    var player = this;
+
+    originals.Player.loadTech_.apply(player, arguments);
+
+    var tech = this.tech_;
+    var proto = tech.constructor.prototype;
+
+    this.polyfilledTechKeys_ = Object.keys(proto);
```

Comment:
why not loop through the keys that are directly on the `tech`? It's theoretically possible there are stuff there that aren't on the prototype.